lib/experimental/fonts/font-library/class-wp-font-collection.php

```diff
@@ -21,41 +21,125 @@
+
+	/**
+	 * The name of the font collection.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var string
+	 */
+	private $name;
+
+	/**
+	 * Description of the font collection.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var string
+	 */
+	private $description;
+
+	/**
+	 * Source of the font collection.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var string
+	 */
+	private $src;
+
+	/**
+	 * Array of font families in the collection.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var array
+	 */
+	private $font_families;
+
+	/**
+	 * Categories associated with the font collection.
 	 *
 	 * @since 6.5.0
 	 *
 	 * @var array
 	 */
-	private $config;
+	private $categories;
+
 
 	/**
 	 * WP_Font_Collection constructor.
 	 *
 	 * @since 6.5.0
 	 *
 	 * @param array $config Font collection config options.
-	 *  See {@see wp_register_font_collection()} for the supported fields.
-	 * @throws Exception If the required parameters are missing.
+	 *  {
+	 *      @type string $slug        The font collection's unique slug.
+	 *      @type string $name        The font collection's name.
+	 *      @type string $description The font collection's description.
+	 *      @type string $src         The font collection's source.
+	 *      @type array  $font_families An array of font families in the font collection.
+	 *      @type array  $categories The font collection's categories.
+	 *  }
 	 */
 	public function __construct( $config ) {
-		if ( empty( $config ) || ! is_array( $config ) ) {
-			throw new Exception( 'Font Collection config options is required as a non-empty array.' );
-		}
+		$this->is_config_valid( $config );
+
+		$this->slug          = $config['slug'] ?? '';
+		$this->name          = $config['name'] ?? '';
+		$this->description   = $config['description'] ?? '';
+		$this->src           = $config['src'] ?? '';
+		$this->font_families = $config['font_families'] ?? array();
+		$this->categories    = $config['categories'] ?? array();
+	}
 
-		if ( empty( $config['slug'] ) || ! is_string( $config['slug'] ) ) {
-			throw new Exception( 'Font Collection config slug is required as a non-empty string.' );
+	/**
+	 * Checks if the font collection config is valid.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param array $config Font collection config options.
+	 *  {
+	 *      @type string $slug        The font collection's unique slug.
+	 *      @type string $name        The font collection's name.
+	 *      @type string $description The font collection's description.
+	 *      @type string $src         The font collection's source.
+	 *      @type array  $font_families An array of font families in the font collection.
+	 *      @type array  $categories The font collection's categories.
+	 *  }
+	 * @return bool True if the font collection config is valid and false otherwise.
+	 */
+	public static function is_config_valid( $config ) {
+		if ( empty( $config ) || ! is_array( $config ) ) {
+			_doing_it_wrong( __METHOD__, 'Font Collection config options are required as a non-empty array.', 'Your_Version_Number' );
```

Comment:
Make sure each of the `_doing_it_wrong` calls here

- Translates the error message
- Uses the full WP version number (`6.5.0`)

```suggestion
			_doing_it_wrong( __METHOD__, __( 'Font Collection config options are required as a non-empty array.', 'gutenberg' ), '6.5.0' );
```

Reply:
done